### PR TITLE
Add js task runner

### PR DESCRIPTION
**Pull request by owencjones**
**Requires orchestrated merge so don't merge for now**
# Motivation and Context
Many of the ends we intend to reach with frontend improvements require the running of tasks, this introduces the task running frameworks.

# What has changed
* Added `node.js`, `npm` setup, and a `gulp` task framework
* Added tasks for every currently intended task to be implemented
* Implemented tasks:
    * CSS Linting
    * SCSS -> CSS compilation
    * JS linting using ESLint
* Other tasks raise a specfic error to point out that they are not yet implemented.

# How to test?
* Run the solution, ensure it works
* Run `gulp scsscompile` to confirm it runs (having followed gulp installation in README)
* Run `docker build .` from the repo to ensure it builds.

# Links
https://trello.com/c/1RVgxFl4

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ language: python
 sudo: required
 dist: trusty
 python: '3.6'
+node: '10'
```
Comment (owencjones) [inline .travis.yml]: Worth noting - I don't know if this works.  Part of the reason for creating the PR was to force it to run in travis.

File: gulp/GulpError.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class GulpError extends Error {
+    constructor(message, fatal = true) {
+        super();
+        this.message = message;
+        this.fatal = fatal;
```
Comment (owencjones) [inline gulp/GulpError.js]: Currently unused - but later tasks will use this to defined whether a GulpError should be build-breaking or not.

File: response_operations_ui/assets/scss/includes/_overrides.scss [hunk 3]
```diff
@@ -0,0 +1,343 @@
+html{
```
Comment (owencjones) [inline response_operations_ui/assets/scss/includes/_overrides.scss]: Basically no need to review this - it's copied across to a new location, but is old code.  I will be refactoring it later.

File: .stylelintrc.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        "selector-pseudo-element-colon-notation": "double",
+        "selector-no-vendor-prefix": true,
+        "media-feature-range-operator-space-before": "always",
+        "media-feature-range-operator-space-after": "always",
+        "media-feature-parentheses-space-inside": "never",
+        "media-feature-name-no-vendor-prefix": true,
+        "media-feature-colon-space-before": "always",
+        "media-feature-colon-space-after": "always"
+    }
+}
```
Comment (NickGrantONS) [inline .stylelintrc.json]: Where did we get this standard from?

Comment (owencjones) [reply .stylelintrc.json]: It's the default generated.  We'll adapt it to not be too intrusive over time.

**Pull request merged**